astropy/utils/console.py

```diff
@@ -631,10 +637,17 @@ def _update_ipython_widget(self, value=None):
         # if none exists.
         if not hasattr(self, '_widget'):
             # Import only if an IPython widget, i.e., widget in iPython NB
-            from IPython.html import widgets
+            try:
+                from ipywidgets import widgets
+            except ImportError:
+                from IPython.html import widgets
             from IPython.display import display
 
-            self._widget = widgets.FloatProgressWidget()
+            try:
+                self._widget = widgets.FloatProgress()
+            except AttributeError:
+                self._widget = widgets.FloatProgressWidget()
```

Comment:
If you want a closer look at how I'm handling the import posssibilites in reducer look here: https://github.com/mwcraig/reducer/blob/ipython-3/reducer/image_browser.py#L18

The version helper is here: https://github.com/mwcraig/reducer/blob/ipython-3/reducer/ipython_version_helper.py